src/smolagents/models.py

```diff
@@ -180,30 +186,53 @@ def remove_stop_sequences(content: str, stop_sequences: List[str]) -> str:
 def get_clean_message_list(
     message_list: List[Dict[str, str]],
     role_conversions: Dict[MessageRole, MessageRole] = {},
+    convert_images_to_image_urls: bool = False,
+    flatten_messages_as_text: bool = False,
 ) -> List[Dict[str, str]]:
     """
     Subsequent messages with the same role will be concatenated to a single message.
 
     Args:
-        message_list (`List[Dict[str, str]]`): List of chat messages.
+        message_list (`list[dict[str, str]]`): List of chat messages.
+        role_conversions (`dict[MessageRole, MessageRole]`, *optional* ): Mapping to convert roles.
+        convert_images_to_image_urls (`bool`, default `False`): Whether to convert images to image URLs.
+        flatten_messages_as_text (`bool`, default `False`): Whether to flatten messages as text.
     """
     final_message_list = []
     message_list = deepcopy(message_list)  # Avoid modifying the original list
-    for message in message_list:
-        # if not set(message.keys()) == {"role", "content"}:
-        #     raise ValueError("Message should contain only 'role' and 'content' keys!")
 
+    for message in message_list:
         role = message["role"]
         if role not in MessageRole.roles():
             raise ValueError(f"Incorrect role {role}, only {MessageRole.roles()} are supported for now.")
 
         if role in role_conversions:
             message["role"] = role_conversions[role]
+        # encode images if needed
+        if isinstance(message["content"], list):
+            for i, element in enumerate(message["content"]):
+                if element["type"] == "image":
+                    assert not flatten_messages_as_text, f"Cannot use images with {flatten_messages_as_text=}"
+                    if convert_images_to_image_urls:
+                        message["content"][i] = {
+                            "type": "image_url",
+                            "image_url": {"url": make_image_url(encode_image_base64(element["image"]))},
+                        }
+                    else:
+                        message["content"][i]["image"] = encode_image_base64(element["image"])
 
         if len(final_message_list) > 0 and message["role"] == final_message_list[-1]["role"]:
-            final_message_list[-1]["content"] += "\n=======\n" + message["content"]
+            assert isinstance(message["content"], list), "Error: wrong content:" + str(message["content"])
+            if flatten_messages_as_text:
+                final_message_list[-1]["content"] += message["content"][0]["text"]
+            else:
+                final_message_list[-1]["content"] += message["content"]
```

Comment:
I think these are inverted:
```suggestion
            if flatten_messages_as_text:
                final_message_list[-1]["content"] += message["content"]
            else:
                final_message_list[-1]["content"] += message["content"][0]["text"]
```